### PR TITLE
POPR-155 Create stepper

### PR DESCRIPTION
**Pull request by pamcontin**
# Related Task
- https://app.clickup.com/t/459857/PROP-155

# Proposed Changes
- Create in UI folder components Stepper and StepChild

# Amplify Preview
- https://pr-XXXXX.d3gl17ubh0iihv.amplifyapp.com

# Additional Info (Optional)
- Anything that reviewers need to be aware of

# How to test this?
- Use it in app with an array of objects with the requirements (label, description and content)

# Task Checklist
- [x] All new features have been documented
- [x] Conventions and Acceptance criteria have been met
- [x] Code is free of warnings
- [x] ClickUp task has been set to Awaiting Review

### CODE REVIEW

File: frontend/ui/src/components/Stepper/StepChild.tsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  stepLength: number;
+}> = ({ step, handleNext, handleBack, index, stepLength }) => {
+  return (
+    <>
+      <StepLabel
+        optional={
+          index === 2 ? (
+            <Typography variant="caption">Last step</Typography>
+          ) : null
+        }
```
Comment (AgusPk) [inline frontend/ui/src/components/Stepper/StepChild.tsx]: what does it represent index === 2?

Comment (pamcontin) [reply frontend/ui/src/components/Stepper/StepChild.tsx]: Oh it doesn't have to be like that, it represent the last step, but it's wrong that it's hardcoded, i will change it now, thanks for notice it!

File: frontend/ui/src/components/Stepper/StepperButtons.tsx [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+      <div>
+        <Button
+          disabled={index === 0}
+          onClick={handleBack}
+          sx={{ mt: 1, mr: 1 }}
+        >
+          Back
+        </Button>
+        <Button variant="contained" onClick={handleNext} sx={{ mt: 1, mr: 1 }}>
+          {index === stepLength - 1 ? "Save" : "Continue"}
```
Comment (AgusPk) [inline frontend/ui/src/components/Stepper/StepperButtons.tsx]: just a matter of taste, can we move this to a constant?

Comment (pamcontin) [reply frontend/ui/src/components/Stepper/StepperButtons.tsx]: of course! i'll change it now

File: frontend/ui/src/components/Stepper/StepChild.tsx [hunk 3]
```diff
@@ -0,0 +1,31 @@
+import React, { FC } from "react";
+import { Step, StepContent, StepLabel, Typography } from "@mui/material";
+import { StepperButtons } from "./StepperButtons";
+
+export const StepChild: FC<{
```
Comment (rosariosm) [inline frontend/ui/src/components/Stepper/StepChild.tsx]: Move this definition to an interface
```
interface Step = {
    label: string;
    description: string;
    content: JSX.Element;
}

interface StepChildProps = {
  step: Step;
  handleNext: () => void;
  handleBack: () => void;
  index: number;
  stepLength: number;
}

... 
export const StepChild: FC<StepChildProps> = ({step, handleNext, ...}) =>  {...}

```

File: frontend/ui/src/components/Stepper/StepperButtons.tsx [hunk 4]
```diff
@@ -0,0 +1,28 @@
+import React, { FC } from "react";
+import { Button, Box } from "@mui/material";
+
+export const StepperButtons: FC<{
```
Comment (rosariosm) [inline frontend/ui/src/components/Stepper/StepperButtons.tsx]: Move this to an interface

```
interface StepperButtonsProps = {
handleNext: () => void;
  handleBack: () => void;
  index: number;
  stepLength: number;
}

export const StepperButtons: FC<StepperButtonsProps> = ({ handleNext, handleBack, index, stepLength }) => {...}
```

**Review by rosariosm — CHANGES_REQUESTED** [review]

**Review by rosariosm — APPROVED** [review]

**Pull request merged**